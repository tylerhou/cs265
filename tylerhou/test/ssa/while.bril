@main {
  a: int = const 10;
.while.cond:
  zero: int = const 0;
  is_term: bool = eq a zero;
  br is_term .while.finish .while.body;
.while.body:
  one: int = const 1;
  a: int = sub a one;
  jmp .while.cond;
.while.finish:
  print a;
}
