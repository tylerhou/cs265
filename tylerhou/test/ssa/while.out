@main {
.__b1:
  a0: int = const 10;
  jmp .while.cond(a0);
.while.cond(a1: int):
  zero: int = const 0;
  is_term: bool = eq a1 zero;
  br is_term .while.finish(a1) .while.body(a1);
.while.body(a2: int):
  one: int = const 1;
  a3: int = sub a2 one;
  jmp .while.cond(a3);
.while.finish(a4: int):
  print a4;
  ret;
}
